apps/system/init/init.c

```diff
@@ -157,5 +167,20 @@ int preapp_start(int argc, char *argv[])
 	}
 #endif
 
+/***********************************************************************************
+ *	current preapp_start does the up_cxxinitialize which initializes the
+ *	static constructors. All the tasks and threads created by preapp refer to
+ *	these constructors. To make sure that c++ test cases from tash runs properly,
+ *	we need to keep preapp/constructor initializations alive with below changes.
+ **********************************************************************************/
+#ifdef CONFIG_HAVE_CXXINITIALIZE
+	while (sem_wait(&sem) != 0) {
+		/* The only case that an error should occur here is if the wait was
+		 * awakened by a signal.
+		 */
+		if (errno == EINTR)
+			printf("awakened by signal..\n");
+		}
```

Comment:
Please remove a tab (wrong indentation) and could you use brackets for if? We guide using them even it has single statement.